src/main/java/com/petrych/screenshotter/service/ScreenshotServiceImpl.java

```diff
@@ -132,10 +140,18 @@ public void update(String urlString) throws MalformedURLException {
 		} else {
 			UrlUtil.isUrlValid(urlString);
 			// update if exists
-			new ScreenshotMaker(getStorageLocation()).createFromUrl(urlString);
+			ArrayList<String> fileNames = (ArrayList<String>) findFileNamesByUrl(urlString);
+			if (fileNames.isEmpty()) {
+				return;
+			}
+			
+			String fileNameToupdate = fileNames.get(0);
+			
+			String screenshotName = new ScreenshotMaker(getStorageLocation()).createScreenshotWithNameAndFile(urlString,
+			                                                                                                  fileNameToupdate);
 			
 			Screenshot screenshot = ((ArrayList<Screenshot>) screenshotRepo
-					.findByNameContaining(fileNameToSearchFor)).get(0);
+					.findByNameContaining(screenshotName)).get(0);
 			screenshot.setDateTimeCreated(LocalDateTime.now());
 			
 			screenshotRepo.save(screenshot);
```

Comment:
I don't fully understand what this code block is doing. Perhaps we can take this in a call? :)